src/packages/end-to-end/src/__tests__/api/auth/acl/EXOGW-416.test.ts

```diff
@@ -0,0 +1,274 @@
+			);
+		}
+
+		// create the task, tag, and task_tags tables
+		await em
+			.getConnection()
+			.execute(
+				'CREATE TABLE `Task` (`id` TEXT NOT NULL, `user_id` TEXT NOT NULL, PRIMARY KEY (`id`))'
+			);
+		await em.getConnection().execute('CREATE TABLE `Tag` (`id` TEXT NOT NULL, PRIMARY KEY (`id`))');
+		await em
+			.getConnection()
+			.execute(
+				'CREATE TABLE `task_tags` (`orm_task_id` TEXT NOT NULL, `orm_tag_id` TEXT NOT NULL, PRIMARY KEY (`orm_task_id`, `orm_tag_id`))'
+			);
+
+		const loginResponse = await graphweaver.executeOperation<{
+			loginPassword: { authToken: string };
+		}>({
+			query: gql`
+				mutation loginPassword($username: String!, $password: String!) {
+					loginPassword(username: $username, password: $password) {
+						authToken
+					}
+				}
+			`,
+			variables: {
+				username: 'test',
+				password: 'test123',
+			},
+		});
+
+		assert(loginResponse.body.kind === 'single');
+		expect(loginResponse.body.singleResult.errors).toBeUndefined();
+
+		token = loginResponse.body.singleResult.data?.loginPassword?.authToken;
+		expect(token).toContain('Bearer ');
+	});
+
+	afterAll(async () => {
+		// delete the task, tag, and task_tags tables
+		await em?.getConnection().execute('DROP TABLE `Task`');
+		await em?.getConnection().execute('DROP TABLE `Tag`');
+		await em?.getConnection().execute('DROP TABLE `task_tags`');
+
+		await em?.getConnection().close();
+	});
+
+	test('Should only return tasks that the user has access to when asking for tags', async () => {
+		const spyOnArtistDataProvider = jest.spyOn(taskProvider, 'findByRelatedId');
+		// create 3 tasks for our user and another 3 for a different user
+		const task1ForAuthenticatedUser = new OrmTask('task1ForAuthenticatedUser', user.id);
+		const task2ForAuthenticatedUser = new OrmTask('task2ForAuthenticatedUser', user.id);
+		const task3ForAuthenticatedUser = new OrmTask('task3ForAuthenticatedUser', user.id);
+		const task1ForOtherUser = new OrmTask('task1ForOtherUser', 'another user');
+		const task2ForOtherUser = new OrmTask('task2ForOtherUser', 'another user');
+		const task3ForOtherUser = new OrmTask('task3ForOtherUser', 'another user');
+
+		// create 3 tags, 1 with a mix of tasks from both users, 1 with tasks from the authenticated user, and 1 with tasks from the other user
+		const mixTag = new OrmTag('mixTag');
+		mixTag.tasks.add(task1ForAuthenticatedUser);
+		mixTag.tasks.add(task2ForOtherUser);
+		const tagForAuthenticatedUser = new OrmTag('tagForAuthenticatedUser');
+		tagForAuthenticatedUser.tasks.add(task1ForAuthenticatedUser);
+		tagForAuthenticatedUser.tasks.add(task2ForAuthenticatedUser);
+		tagForAuthenticatedUser.tasks.add(task3ForAuthenticatedUser);
+		const tagForOtherUser = new OrmTag('tagForOtherUser');
+		tagForOtherUser.tasks.add(task1ForOtherUser);
+		tagForOtherUser.tasks.add(task2ForOtherUser);
+		tagForOtherUser.tasks.add(task3ForOtherUser);
+
+		await em?.persistAndFlush([
```

Comment:
```suggestion
		await em!.persistAndFlush([
```

Probably makes sense for this to throw if it can't run because `em` is undefined.